In [1]:
import numpy as np

In [7]:
from math import factorial

In [8]:
def soch (n, k):
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n-k))

##### 1. Из колоды в 52 карты извлекаются случайным образом 4 карты.<br> a) Найти вероятность того, что все карты – крести.<br> б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

a) общее число исходов = $ C_{52}^{4} $, число исходов с 4 трефами = $ C_{13}^{4}$

In [47]:
(soch(13, 4) / soch(52, 4))*100

0.26410564225690275

Ответ: 0.26%

б) общее число исходов = $ C_{52}^{4} $, число исходов без туза = $ C_{48}^{4} $

In [51]:
(1 - (soch(48, 4) / soch(52, 4)))*100

28.12632745405854

Ответ: 28.13%

##### 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно.<br> Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Общее число вариантов кода $ C_{10}^{3} $, всего 1 попытка на ввод

In [53]:
(1 / soch(10, 3))*100

0.8333333333333334

Ответ: 0.83%

##### 3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали.<br> Какова вероятность того, что все извлеченные детали окрашены?

Общее число исходов = $ C_{15}^{9} $, число благоприятствующих исходов = $ C_{9}^{3} $

In [54]:
(soch(9, 3) / soch(15, 3))*100

18.461538461538463

Ответ: 18.46%

##### 4. В лотерее 100 билетов. Из них 2 выигрышных.<br>Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

Общее число исходов = $ C_{100}^{2} $, число благоприятствующих исходов = $ C_{2}^{2} $ == 1

In [55]:
(1 / soch(100, 2))*100

0.0202020202020202

Ответ: 0.02%